src/OpenTracing/Util/AsyncLocalScope.cs

```diff
@@ -0,0 +1,43 @@
+namespace OpenTracing.Util
+{
+    /// <summary>
+    /// The <see cref="AsyncLocalScope"/> is a simple <see cref="IScope"/> implementation
+    /// that relies on C#'s AsyncLocal/CallContext storage primitive.
+    /// </summary>
+    /// <seealso cref="IScopeManager"/>
+    public class AsyncLocalScope : IScope
```

Comment:
This relates to the other review comment I made regarding the APIs. Using this structure removes the ability for any implementation to expect that the Scope they get back will be the one they pushed out of their APIs, which can cause some weird issues.

Do not have a good suggestion at this time, so this is not a comment to request change but just to start your brain pondering over solutions.

Reply:
Could you clarify this? I do not understand what you mean.